packages/ckeditor5-ui/src/editorui/poweredby.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/editorui/poweredby
+ */
+
+import type { Editor } from '@ckeditor/ckeditor5-core';
+import { DomEmitterMixin, type PositionOptions, type Locale, type Rect } from '@ckeditor/ckeditor5-utils';
+import BalloonPanelView from '../panel/balloon/balloonpanelview';
+import IconView from '../icon/iconview';
+import View from '../view';
+
+import poweredByIcon from '../../theme/icons/project-logo.svg';
+
+const POWERED_BY_VIEW_SYMBOL = Symbol( '_poweredByView' );
+const POWERED_BY_BALLOON_SYMBOL = Symbol( '_poweredByBalloon' );
+const ICON_WIDTH = 53;
+const ICON_HEIGHT = 10;
+const CORNER_OFFSET = 5;
+const NARROW_ROOT_WIDTH_THRESHOLD = 250;
+const OFF_THE_SCREEN_POSITION = {
+	top: -9999999,
+	left: -9999999,
+	name: 'invalid'
+};
+
+/**
+ * A helper that enables the "powered by" feature in the editor and renders a link to the project's
+ * webpage next to the bottom of the editing root when the editor is focused.
+ *
+ * The helper uses a {@link module:ui/panel/balloon/balloonpanelview~BalloonPanelView balloon panel}
+ * to position the link with the logo.
+ *
+ * @private
+ */
+export default class PoweredBy extends DomEmitterMixin() {
+	/**
+	 * A reference to the view displaying a link with a label and a project logo.
+	 */
+	private [ POWERED_BY_VIEW_SYMBOL ]: PoweredByView | null;
+
+	/**
+	 * A reference to the balloon panel hosting and positioning the "powered by" view.
+	 */
+	private [ POWERED_BY_BALLOON_SYMBOL ]: BalloonPanelView | null;
```

Comment:
These make it quite impossible to test the feature. I'm wondering if we could make these properties public since CSS classes are already there allowing manipulation or disabling of the feature.